src/bindings/fluid_ladspa.c

```diff
@@ -22,1107 +22,1288 @@
-      };
-
-      /*
-       *
-       * Connect the port
-       *
-       */
-
-      L(fluid_ostream_printf(out,"Connecting %i",CurrentPort_Index));
-      CurrentPluginDescriptor->connect_port
-	(CurrentPlugin,
-	 CurrentPort_Index,
-	 Current_Node->buf
-	 );
-      CurrentPlugin_PortConnected[CurrentPort_Index]++;
-
-    }; /* While Tokensequence (more connections) */
-
-    /*
-     *
-     * Check for left-over tokens
-     *
-     */
+    /* Run each plugin on a block of data */
+    for (i = 0; i < fx->num_plugins; i++)
+    {
+        fx->plugins[i]->desc->run(fx->plugins[i]->handle, FLUID_BUFSIZE);
+    };
 
-    if (TokenSequence[TokenCount]){
-      char * T1="";char * T2="";char * T3="";
-      if (TokenSequence[TokenCount]){T1=TokenSequence[TokenCount];};
-      if (TokenSequence[TokenCount+1]){T2=TokenSequence[TokenCount+1];};
-      if (TokenSequence[TokenCount+2]){T3=TokenSequence[TokenCount+2];};
-    fluid_ostream_printf(out, "***Error012***\n"
-	       "Leftover tokens: %s %s %s...\n",T1,T2,T3);
-      fluid_LADSPA_clear(FxUnit);
-      return(PrintErrorMessage);
+    /* Copy the data from the output nodes back to the synth. */
+    for (i = 0; i < fx->audio_channels; i++)
+    {
+        node_to_buffer(fx->nodes[n++], left_buf[i]);
+        node_to_buffer(fx->nodes[n++], right_buf[i]);
     };
+};
 
-    /*
-     *
-     * Check, that all plugin ports are connected
-     *
-     */
-    L(fluid_ostream_printf(out,"Checking left-over ports"));
-    assert(CurrentPluginDescriptor);
-    for (PortCount=0; PortCount<CurrentPluginDescriptor->PortCount; PortCount++){
-      assert(CurrentPlugin_PortConnected[PortCount] <=1);
-      if (CurrentPlugin_PortConnected[PortCount] !=1){
-    fluid_ostream_printf(out, "***Error013***\nPlugin: %s. Port %s is unconnected!\n",PluginLabel, CurrentPluginDescriptor->PortNames[PortCount]);
-	fluid_LADSPA_clear(FxUnit);
-	return(PrintErrorMessage);
-      };
+/**
+ * Creates the nodes to get data from FluidSynth into the LADSPA effects
+ * and back out again to FluidSynth
+ *
+ * The nodes are named from LADSPAs perspective. So the "in*" and "send*" nodes carry audio data
+ * from FluidSynth into LADSPA, "out*" nodes carry audio data from LADSPA back into FluidSynth.
+ *
+ * System nodes are referenced from the fx->nodes array, just like user created nodes. The system
+ * nodes are always the first in the array. The order of nodes is:
+ *  in1_L, in1_R, ..., send1_L, send1_R, ..., out1_L, out1_R, ..., [user created nodes ...]
+ *
+ * @param fx LADSPA fx instance
+ * @return FLUID_OK on success, otherwise FLUID_FAILED
+ */
+static int create_input_output_nodes(fluid_ladspa_fx_t *fx)
+{
+    char name[99];
+    int i;
+
+    /* These nodes transport the main audio generated by FluidSynth into the
+     * LADSPA effects unit. Create left and right input nodes for each audio group. */
+    for (i = 0; i < fx->audio_groups; i++)
+    {
+        FLUID_SPRINTF(name, "in%i_L", (i + 1));
```

Comment:
use FLUID_S**N**PRINTF pls

`FLUID_SNPRINTF(name, sizeof(name), "in%i_L", (i + 1));`

Reply:
Ack, fixed